rocketmq-connect-runtime/src/main/java/org/apache/rocketmq/connect/runtime/connectorwrapper/Worker.java

```diff
@@ -900,6 +910,21 @@ private void startTask(Map<String, List<ConnectKeyValue>> newTasks) throws Excep
         }
     }
 
+    private void redressRunningStatus(WorkerTask workerTask) {
+        TaskStatus taskStatus = stateManagementService.get(workerTask.id);
+        if (taskStatus != null && taskStatus.getState() != RUNNING && taskStatus.getState() != PAUSED) {
+            ConnectorStatus connectorStatus = stateManagementService.get(workerTask.id.connector());
+            TaskStatus newTaskStatus;
+            if (null != connectorStatus && connectorStatus.getState() == PAUSED) {
+                newTaskStatus = new TaskStatus(workerTask.id, PAUSED, workerConfig.getWorkerId(), System.currentTimeMillis());
+            } else {
+                newTaskStatus = new TaskStatus(workerTask.id, RUNNING, workerConfig.getWorkerId(), System.currentTimeMillis());
```

Comment:
IMO, It is better to set the task status to running when the connector is in the running state. If the connector is failed, the task is also set to running at present

Reply:
WorkerConnector seems to have a problem with the need for state error correction logic？